Add '-set_exit_status=1' to golint command

`golint` requires the option `-set_exit_status=1` to be set in order to fail the command. Without it, we ended up merging PRs that are not passing the linting checks:

```
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/account/main.go:48:1: exported function JaegerServiceAccountFor should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1/jaeger_types.go:157:6: exported type ElasticsearchSpec should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:15:2: exported const ServiceAccountName should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:69:6: exported type ElasticsearchStorageSpec should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:85:6: exported type ElasticsearchNodeUpgradeStatus should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:90:6: exported type ElasticsearchUpgradePhase should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:93:2: exported const NodeRestarting should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:104:6: exported type UpgradeStatus should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:107:2: exported const UnderUpgradeTrue should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:111:6: exported type ElasticsearchRequiredAction should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:114:2: exported const ElasticsearchActionRollingRestartNeeded should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:122:6: exported type ElasticsearchNodeRole should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:125:2: exported const ElasticsearchRoleClient should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:130:6: exported type ShardAllocationState should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:133:2: exported const ShardAllocationTrue should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:146:6: exported type PodStateMap should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:148:6: exported type PodStateType should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:151:2: exported const PodStateTypeReady should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:156:6: exported type ManagementState should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:159:2: comment on exported const ManagementStateManaged should be of the form "ManagementStateManaged ..."
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:162:2: comment on exported const ManagementStateUnmanaged should be of the form "ManagementStateUnmanaged ..."
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:191:6: exported type ConditionStatus should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:194:2: exported const ConditionTrue should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:199:6: exported type ClusterEvent should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:202:2: exported const ScaledDown should have comment (or a comment on this block) or be unexported
```

For this to work, the following needs to be fixed. @pavolloffay, most of those are related to ES, would you please document them?

```
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/account/main.go:48:1: exported function JaegerServiceAccountFor should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch_secrets.go:91:1: comment on exported function CreateESCerts should be of the form "CreateESCerts ..."
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:15:2: exported const ServiceAccountName should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:69:6: exported type ElasticsearchStorageSpec should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:85:6: exported type ElasticsearchNodeUpgradeStatus should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:90:6: exported type ElasticsearchUpgradePhase should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:93:2: exported const NodeRestarting should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:104:6: exported type UpgradeStatus should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:107:2: exported const UnderUpgradeTrue should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:111:6: exported type ElasticsearchRequiredAction should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:114:2: exported const ElasticsearchActionRollingRestartNeeded should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:122:6: exported type ElasticsearchNodeRole should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:125:2: exported const ElasticsearchRoleClient should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:130:6: exported type ShardAllocationState should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:133:2: exported const ShardAllocationTrue should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:146:6: exported type PodStateMap should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:148:6: exported type PodStateType should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:151:2: exported const PodStateTypeReady should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:156:6: exported type ManagementState should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:159:2: comment on exported const ManagementStateManaged should be of the form "ManagementStateManaged ..."
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:162:2: comment on exported const ManagementStateUnmanaged should be of the form "ManagementStateUnmanaged ..."
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:191:6: exported type ConditionStatus should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:194:2: exported const ConditionTrue should have comment (or a comment on this block) or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:199:6: exported type ClusterEvent should have comment or be unexported
/mnt/storage/jpkroehling/Projects/src/github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1alpha1/types.go:202:2: exported const ScaledDown should have comment (or a comment on this block) or be unexported
```

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ STORAGE_NAMESPACE ?= "${shell kubectl get sa default -o jsonpath='{.metadata.nam
 ES_OPERATOR_NAMESPACE = openshift-logging
 
 LD_FLAGS ?= "-X $(VERSION_PKG).version=$(OPERATOR_VERSION) -X $(VERSION_PKG).buildDate=$(VERSION_DATE) -X $(VERSION_PKG).defaultJaeger=$(JAEGER_VERSION)"
-PACKAGES := $(shell go list ./cmd/... ./pkg/...)
+PACKAGES := $(shell go list ./cmd/... ./pkg/... |  grep -v elasticsearch/v1)
 
 .DEFAULT_GOAL := build
 
@@ -41,7 +41,7 @@ format:
 .PHONY: lint
 lint:
 	@echo Linting...
-	@golint $(PACKAGES)
+	@golint -set_exit_status=1 $(PACKAGES)
 	@gosec -quiet -exclude=G104 $(PACKAGES) 2>/dev/null
 
 .PHONY: build

--- a/pkg/account/main.go
+++ b/pkg/account/main.go
@@ -45,6 +45,7 @@ func getMain(jaeger *v1.Jaeger) *corev1.ServiceAccount {
 	}
 }
 
+// JaegerServiceAccountFor prints service name for Jaeger instance
 func JaegerServiceAccountFor(jaeger *v1.Jaeger) string {
 	return fmt.Sprintf("%s", jaeger.Name)
 }

--- a/pkg/storage/elasticsearch_secrets.go
+++ b/pkg/storage/elasticsearch_secrets.go
@@ -88,8 +88,10 @@ func ESSecrets(jaeger *v1.Jaeger) []corev1.Secret {
 	}
 }
 
-// The secrets are pulled back to FS in case of operator restart
-// The script checks if secrets are expired or need to be regenerated
+// CreateESCerts creates certificates for elasticsearch, jaeger and curator
+// The cert generation is done by shell script. If the certificates are not present
+// on the filesystem the operator injects them from secrets - this allows operator restarts.
+// The script also re-generates expired certificates.
 func CreateESCerts(jaeger *v1.Jaeger, existingSecrets []corev1.Secret) error {
 	err := extractSecretsToFile(jaeger, existingSecrets, masterSecret, esSecret, jaegerSecret, curatorSecret)
 	if err != nil {

